File: app/src/main/java/com/example/salvatore/sddialogview/CustomDialog.java
package com.example.salvatore.sddialogview;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.RelativeLayout;
import android.widget.Toast;

import com.sysdata.sddialog.SDDialogView;

import java.lang.ref.WeakReference;

/**
 * Created by Salvatore on 28/06/2018.
 */

public class CustomDialog extends RelativeLayout implements View.OnClickListener, SDDialogView.Compatible {
    private SDDialogView parent;
    private Button mCloseBtn;
    private Button mFirstBtn;
    private Button mSecondBtn;

    public CustomDialog(Context context) {
        super(context);
        init(context, null);
    }

    public CustomDialog(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public CustomDialog(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
        View view = inflater.inflate(R.layout.custom_dialog, this, true);
        initView(view);
    }

    private void initView(View view) {
        mCloseBtn = view.findViewById(R.id.close_btn);
        mFirstBtn = view.findViewById(R.id.first_btn);
        mSecondBtn = view.findViewById(R.id.second_btn);
        mCloseBtn.setOnClickListener(this);
        mFirstBtn.setOnClickListener(this);
        mSecondBtn.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if(view == mCloseBtn){
            parent.closeDialog(0);
        }
        else if(view == mFirstBtn){
            Toast.makeText(getContext(), "first", Toast.LENGTH_SHORT).show();
        }
        else if(view == mSecondBtn){
            Toast.makeText(getContext(), "second", Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public void onBindParentView(SDDialogView parent) {
        this.parent = parent;
    }

    @Override
    public void onUnbindParentView() {
        this.parent = null;
    }
}

File: KTSDDialogView/app/src/main/java/com/example/salvatore/ktsddialogview/CustomDialog.kt
package com.example.salvatore.ktsddialogview

import android.app.Activity
import android.content.Context
import android.util.AttributeSet
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.widget.RelativeLayout
import android.widget.Toast
import com.sysdata.kt.sddialog.SDDialogView
import kotlinx.android.synthetic.main.custom_dialog.view.*

/**
 * Created by Salvatore on 06/07/2018.
 */
class CustomDialog(context: Context?, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : RelativeLayout(context, attrs, defStyleAttr), View.OnClickListener, SDDialogView.Compatible {

    private var parent: SDDialogView? = null

    init {
        context?.let {
            val inflater = it.getSystemService(Activity.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            inflater.inflate(R.layout.custom_dialog, this, true)
            close_btn.setOnClickListener(this)
            first_btn.setOnClickListener(this)
            second_btn.setOnClickListener(this)
        }
    }

    override fun onClick(v: View?) {
        v?.let {
            when (v) {
                close_btn -> parent?.closeDialog(0)
                first_btn -> Toast.makeText(context, "first", Toast.LENGTH_SHORT).show()
                second_btn -> Toast.makeText(context, "second", Toast.LENGTH_SHORT).show()
                else -> Log.i(this.javaClass.name, " view not handled ")
            }
        }
    }

    override fun onBindParentView(parent: SDDialogView) {
        this.parent = parent
    }

    override fun onUnbindParentView() {
        this.parent = null
    }

}